### ISSUE TITLE
Lastest Docker Version 17.06.0-ce-mac17 (18432) Causes Docker Service Failure (Known Workaround)

### ISSUE THREAD

**Issue by garystafford** [Accenture/adop-docker-compose#230]
The latest version of Docker causes the `docker.service` to fail on Ubuntu host.

```text
Running pre-create checks...
Creating machine...
(adop-server) Launching instance...
Waiting for machine to be running, this may take a few minutes...
Detecting operating system of created instance...
Waiting for SSH to be available...
Detecting the provisioner...
Provisioning with ubuntu(systemd)...
Installing Docker...
Copying certs to the local machine directory...
Copying certs to the remote machine...
Setting Docker configuration on the remote daemon...
Error creating machine: Error running provisioning: ssh command error:
command : sudo systemctl -f start docker
err     : exit status 1
output  : Job for docker.service failed because the control process exited with error code.
See "systemctl status docker.service" and "journalctl -xe" for details.
```

Known fix, posted on Docker Forums (<https://forums.docker.com/t/docker-machine-provisioned-aws-instance-can-not-start-docker-engine/34200/4?u=garystafford)[https://forums.docker.com/t/docker-machine-provisioned-aws-instance-can-not-start-docker-engine/34200/4?u=garystafford>]

To fix ADOP, add `--engine-install-url` line to `quickstart.sh` script:

```bash
        MACHINE_CREATE_CMD="docker-machine create \
                    --driver amazonec2 \
                    --amazonec2-vpc-id ${AWS_VPC_ID} \
                    --amazonec2-zone $VPC_AVAIL_ZONE \
                    --amazonec2-instance-type ${AWS_DOCKER_MACHINE_SIZE} \
                    --amazonec2-root-size ${AWS_ROOT_SIZE:-32} \
                    --engine-install-url=https://web.archive.org/web/20170623081500/https://get.docker.com"
```

I've added line and tested, successfully.

**Comment by anton-kasperovich** [comment]
Looks like this issue already fixed in 0.12.1, could you please just upgrade your Docker-Machine separately (i.e. don't wait when they will make big release of Docker for Mac in like 3 months or so) and test once again, as I don't think that it's a good idea to make workarounds in our stuff just because Docker has temp. bug.
https://github.com/docker/machine/issues/4156#issuecomment-312387395

**Comment by RobertNorthard** [comment]
Closing as per @anton-kasperovich comments.
